src/sso/src/main.rs

```diff
@@ -0,0 +1,316 @@
+                "0.0",
+                &self.session_id,
+                &json!({
+                    "account": account,
+                    "authParameters": self._get_auth_parameters(account, scopes),
+                }),
+            )
+            .await
+            .map_err(|e| Box::new(e) as Box<dyn Error>)
+    }
+
+    async fn get_broker_version(&self) -> Result<Value, Box<dyn Error>> {
+        self.broker
+            .get_linux_broker_version(
+                "0.0",
+                &self.session_id,
+                &json!({
+                    "msalCppVersion": env!("CARGO_PKG_VERSION"),
+                }),
+            )
+            .await
+            .map_err(|e| Box::new(e) as Box<dyn Error>)
+    }
+}
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// Run in interactive mode
+    #[arg(short, long)]
+    interactive: bool,
+
+    /// Account index to use for operations
+    #[arg(short, long, default_value_t = 0)]
+    account: usize,
+
+    /// ssoUrl part of SSO PRT cookie request
+    #[arg(name = "ssoUrl", short, long, default_value_t = SSO_URL_DEFAULT.to_string())]
+    sso_url: String,
+
+    #[command(subcommand)]
+    command: Option<SubCommands>,
+}
+
+#[derive(Subcommand, Debug)]
+enum SubCommands {
+    #[command(name = "getAccounts")]
+    GetAccounts,
+    #[command(name = "getVersion")]
+    GetVersion,
+    #[command(name = "acquirePrtSsoCookie")]
+    AcquirePrtSsoCookie,
+    #[command(name = "acquireTokenSilently")]
+    AcquireTokenSilently,
+}
+
+async fn run_interactive() -> Result<(), Box<dyn Error>> {
+    let args = Args::parse();
+    let ssomib = SsoMib::new().await?;
+    let accounts = ssomib.get_accounts().await?;
+
+    match args.command {
+        Some(SubCommands::GetAccounts) => {
+            println!("{}", serde_json::to_string_pretty(&accounts)?);
+        }
+        Some(SubCommands::GetVersion) => {
+            println!(
+                "{}",
+                serde_json::to_string_pretty(&ssomib.get_broker_version().await?)?
+            );
+        }
+        Some(SubCommands::AcquirePrtSsoCookie) => {
+            let account = &accounts["accounts"][args.account];
+            let cookie = ssomib
+                .acquire_prt_sso_cookie(account, &args.sso_url, None)
+                .await?;
+            println!("{}", serde_json::to_string_pretty(&cookie)?);
+        }
+        Some(SubCommands::AcquireTokenSilently) => {
+            let account = &accounts["accounts"][args.account];
+            let token = ssomib.acquire_token_silently(account, None).await?;
+            println!("{}", serde_json::to_string_pretty(&token)?);
+        }
+        _ => Args::command().print_help()?,
+    }
+    Ok(())
+}
+
+async fn run_as_native_messaging() -> Result<(), Box<dyn Error>> {
+    println!("Running as native messaging instance.");
+    println!("For interactive mode, start with --interactive");
```

Comment:
(facepalm) These messages need to go to stderr, otherwise they're interpreted as corrupted native messages.

Reply:
Funnily enough I fixed that to compare the messages on stdout. That is why it worked for me!

Reply:
Lol! That explains why you got it working when I couldn't. It was just a simple mistake.